### PR TITLE
Update RadioCard content

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

This PR updates the Radio page content following the model Checkbox as a reference.

In a separate commit, I've added a few fixes to the Checkbox content and have consistently ordered sections: group, field, base.

🥺 [Have a look at me, the RadioCard page](https://hds-website-git-alex-ju-radio-card-docs-hashicorp.vercel.app/components/form/radio-card) 

### :link: External links

Jira ticket: [HDS-1289](https://hashicorp.atlassian.net/browse/HDS-1289)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

File: website/docs/components/form/radio-card/partials/code/component-api.md [hunk 1]
```diff
@@ -22,20 +22,26 @@ Here is the API for the `RadioCard` component:
...
   </C.Property>
   <C.Property @name="maxWidth" @type="string" @valueNote="any valid CSS width (%, vw, etc)">
     When used with a `fluid` layout, this parameter will determine the number of cards shown per row (for example `25%` will result in 4 cards). When used with a `fixed` layout, this parameter will preserve the width of the card and wrap cards on multiple rows if necessary.
   </C.Property>
   <C.Property @name="extraAriaDescribedBy" @type="string">
-    An extra ID attribute to be added to the `aria-describedby` HTML attribute. _Notice: by default the `aria-describedby` attribute is automatically generated by the component, using the IDs of the helper text and errors (if they're present); use this argument if you need to pass an extra ID for specific reasons you may have._
+    An additional ID attribute to be added to the `aria-describedby` HTML attribute.
+    <br/><br/>
+    By default, the `aria-describedby` attribute is automatically generated by the component, using the IDs of the helper text and errors (if they’re present); use this argument if you need to pass an extra ID for specific reasons you may have.
```
Comment (jorytindall) [inline website/docs/components/form/radio-card/partials/code/component-api.md:33]:
```suggestion
    By default, the `aria-describedby` attribute is automatically generated by the component using the IDs of the helper text and errors (if they’re present). Use this argument if you need to pass an extra ID for specific reasons you may have.
```

File: website/docs/components/form/radio-card/partials/code/component-api.md [hunk 2]
```diff
@@ -75,30 +79,34 @@ Here is the API for the "group" component:
...
 
 <Doc::ComponentApi as |C|>
   <C.Property @name="<[G].Legend>" @type="yielded component">
     It is an (optional) container that yields its content inside the `<legend>` element. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Legend`](/components/form/base-elements/) component.
   </C.Property>
   <C.Property @name="<[G].HelperText>" @type="yielded component">
-    It is a container that yields its content inside the "helper text" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::HelperText`](/components/form/base-elements/) component. _Notice: the `id` attribute of the element is automatically generated._
+    It is a container that yields its content inside the "helper text" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::HelperText`](/components/form/base-elements/) component.
```
Comment (jorytindall) [inline website/docs/components/form/radio-card/partials/code/component-api.md:97]:
```suggestion
    A container that yields its content inside the "helper text" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::HelperText`](/components/form/base-elements/) component.
```

File: website/docs/components/form/radio-card/partials/code/component-api.md [hunk 3]
```diff
@@ -75,30 +79,34 @@ Here is the API for the "group" component:
...
+    <br/><br/>
+    The `id` attribute of the element is automatically generated.
   </C.Property>
   <C.Property @name="<[G].RadioCard>" @type="yielded component">
     It is used to yield one or more cards inside the group. For details about its API check the `RadioCard` component above.
```
Comment (jorytindall) [inline website/docs/components/form/radio-card/partials/code/component-api.md:102]:
```suggestion
    Used to yield one or more cards inside the group. For details about its API check the `RadioCard` component above.
```

File: website/docs/components/form/radio-card/partials/code/component-api.md [hunk 4]
```diff
@@ -75,30 +79,34 @@ Here is the API for the "group" component:
...
   </C.Property>
   <C.Property @name="<[G].Error>" @type="yielded component">
-    It is a container that yields its content inside the "error" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Error`](/components/form/base-elements/) component. _Notice: the `id` attribute of the `Error` element is automatically generated._
+    It is a container that yields its content inside the "error" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Error`](/components/form/base-elements/) component.
```
Comment (jorytindall) [inline website/docs/components/form/radio-card/partials/code/component-api.md:105]:
```suggestion
    A container that yields its content inside the "error" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Error`](/components/form/base-elements/) component.
```

File: website/docs/components/form/radio-card/partials/code/how-to-use.md [hunk 5]
```diff
@@ -1,8 +1,17 @@
-#### Form::Radio::Group
+## How to use this component
 
-##### Basic use
+This "group" component creates:
```
Comment (jorytindall) [inline website/docs/components/form/radio-card/partials/code/how-to-use.md:3]:
```suggestion
The "group" component creates:
```

**Review by jorytindall — APPROVED** [review]
Small changes suggested

**Review by MelSumner — APPROVED** [review]

**Pull request merged**